### PR TITLE
fix: market data for native tokens with non zero addresses

### PR DESCRIPTION
**Pull request by bergeron**
## **Description**

Draft

When querying the price API, the native token is usually represented by the zero address.  But this is not the case on some chains like polygon, whose native token has a contract `0x0000000000000000000000000000000000001010`.

Depends on: https://github.com/MetaMask/core/pull/4952

[Open in GitHub Codespaces](https://codespaces.new/MetaMask/metamask-extension/pull/28584?quickstart=1)

## **Related issues**

Fixes:

## **Manual testing steps**

(pre-req - you may need to enable the `PORTFOLIO_VIEW=true` flag)

1. Onboard with an SRP that have POL tokens.
2. Connect and switch to Polygon
3. View the POL token on the tokens section on the home page.
    - Does it contain percentages and market data.
5. View the POL asset page.
    - Does it contain the market details view; and percentage sections?

## **Screenshots/Recordings**

| Before | After |
|--------|--------|
| Screenshot 2024-11-22 at 16 42 25 | Screenshot 2024-11-22 at 16 29 38 |
| Screenshot 2024-11-22 at 16 42 41 | Screenshot 2024-11-22 at 16 29 46 | 

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/develop/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -131,6 +131,7 @@
     "attributions:generate": "./development/generate-attributions.sh"
   },
   "resolutions": {
+    "@metamask/assets-controllers": "patch:@metamask-previews/assets-controllers@44.1.0-preview-3487bcdd#.yarn/patches/@metamask-assets-controllers-npm-44.0.0-c223d56176.patch",
```
[the remark this answers is not in the document]
Comment (Prithpal-Sooriya) [reply package.json:135]: TODO - remove this once we release the assets controllers with the new changes.

File: ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx [hunk 2]
```diff
@@ -143,5 +147,7 @@ export const PercentageAndAmountChange = ({
     }
   }
 
+  console.log(formattedValue, formattedValuePrice);
```
Comment (darkwing) [inline ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx:150]: Will eventually need removal.

Comment (Prithpal-Sooriya) [reply ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx:150]: WHOOPS ty for catching this!

Comment (Prithpal-Sooriya) [reply ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx:150]: Removed, thanks for catching this - I was using it to assist writing tests.

Comment (Prithpal-Sooriya) [reply ui/components/multichain/token-list-item/price/percentage-and-amount-change/percentage-and-amount-change.tsx:150]: I'll sweep through to see there are no other logs are added.

File: ui/components/app/wallet-overview/aggregated-percentage-overview.test.tsx [hunk 3]
```diff
@@ -26,20 +27,22 @@ jest.mock('../../../selectors', () => ({
...
-const mockGetShouldHideZeroBalanceTokens =
-  getShouldHideZeroBalanceTokens as jest.Mock;
-
+const mockGetIntlLocale = jest.mocked(getIntlLocale);
+const mockGetCurrentCurrency = jest.mocked(getCurrentCurrency);
+const mockGetPreferences = jest.mocked(getPreferences);
+const mockGetSelectedAccount = jest.mocked(getSelectedAccount);
+const mockGetShouldHideZeroBalanceTokens = jest.mocked(
+  getShouldHideZeroBalanceTokens,
+);
```
Comment (gambinish) [inline ui/components/app/wallet-overview/aggregated-percentage-overview.test.tsx:43]: TIL `jest.mocked` way better than casting to `jest.Mock` like I'm used to 👍

**Review by gambinish — DISMISSED** [review]
LGTM. Pulled down and ran. Populates both the percent change as well as the graph over time in asset details 👍 

If you need another pass after removing the patch, lmk.

File: .yarn/patches/@metamask-assets-controllers-npm-45.0.0-31810ece32.patch
[the remark this answers is not in the document]
Comment (Prithpal-Sooriya) [reply .yarn/patches/@metamask-assets-controllers-npm-45.0.0-31810ece32.patch:1]: Damn, looking at CI, the `$cockatiel` patch might be needed for storybook build. Noted.

File: .yarn/patches/@metamask-assets-controllers-npm-45.1.0-d914c453f0.patch [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+index a13403446a2376d4d905a9ef733941798da89c88..19fd490a5d0446c29ad24cc39465f3984a969406 100644
+--- a/dist/token-prices-service/codefi-v2.mjs
++++ b/dist/token-prices-service/codefi-v2.mjs
+@@ -12,8 +12,7 @@ var __classPrivateFieldGet = (this && this.__classPrivateFieldGet) || function (
+ var _CodefiTokenPricesServiceV2_tokenPricePolicy;
+ import { handleFetch } from "@metamask/controller-utils";
+ import { hexToNumber } from "@metamask/utils";
+-import $cockatiel from "cockatiel";
+-const { circuitBreaker, ConsecutiveBreaker, ExponentialBackoff, handleAll, retry, wrap, CircuitState } = $cockatiel;
++import { circuitBreaker, ConsecutiveBreaker, ExponentialBackoff, handleAll, retry, wrap, CircuitState } from "cockatiel"
```
Comment (Prithpal-Sooriya) [inline .yarn/patches/@metamask-assets-controllers-npm-45.1.0-d914c453f0.patch:11]: This patch is needed for the  `storybook:build` command.
Check the other (deleted) patches. The remaining changes from the old patches are not required anymore.

**Review by gambinish — APPROVED** [review]

**Review by darkwing — APPROVED** [review]

**Pull request merged**